tests/semantic_operator_parsing_test.py

```
@@ -0,0 +1,168 @@
+        
+        # The single mutation prompt asks for just the mutated idea
+        response = """This is a detailed mutation that transforms the original idea using a perspective shift. Instead of focusing on individual productivity, this approach emphasizes community-driven solutions where multiple stakeholders collaborate to achieve shared goals. The implementation involves setting up collaboration platforms, establishing governance structures, and creating incentive mechanisms. Technologies include blockchain for transparency, AI for matching collaborators, and cloud infrastructure for scalability. Expected outcomes include 3x productivity gains through synergy and reduced duplicate efforts."""
+        
+        # For single mutations, the response IS the mutation
+        assert len(response) > 150
+        assert "perspective shift" in response
+        assert "community-driven solutions" in response
+    
+    def test_parse_batch_mutation_response(self):
+        """Test parsing of batch mutation response."""
+        operator = BatchSemanticMutationOperator(llm_provider=None)
```

Ruff (0.12.2)

Local variable `operator` is assigned to but never used

Remove assignment to unused variable `operator`

(F841)